tidy3d/components/types.py

```diff
@@ -215,6 +215,8 @@ def __modify_schema__(cls, field_schema):
 FreqBoundMin = float
 FreqBound = Tuple[FreqBoundMin, FreqBoundMax]
 
+PermittivityComponent = Literal["xx", "xy", "xz", "yx", "yy", "yz", "zx", "zy", "zz", None]
```

Comment:
typically, we just leave out `None` in the `Literal` and handle it separately. 

For example, if the argument is optional

```py
def f(c: PermittivityComponent = None)
```

or you can also do 
```py
comp : typing.Optional[PermittivityComponent]
```